### DISCUSSION TITLE
What is this Library? Coming Soon!

### DISCUSSION THREAD

**Discussion by KevinVandy** [KevinVandy/mantine-react-table#1] [Announcements]
Mantine React Table is an advanced data grid component built with TanStack Table v8 and Mantine. 

This library is forked from [Material React Table](https://www.material-react-table.com/) and builds on its success.

This library should be in beta by the end of January 2023, and possibly release a stable 1.0 in the next couple months.

Most of the work for the library itself is done, pending a rewrite of the advanced filtering features to work with Mantine components. A lot of work still needs to go into the docs and examples, though.

v0.1.0 can be installed, but don't use it for production yet, as it will go through more changes soon.

Docs site is now up: https://www.mantine-react-table.com/

You can see a storybook site showcasing it here: https://www.mantine-react-table.dev/

**Comment by its-monotype** [comment]
Damn, this is definitely cool! Just wanted to try Mantine. And there was just no built-in DataGrid.

**Reply by KevinVandy** [reply]
Hey, just published v0.1.0 and deployed the dev storybook site so people can start seeing it https://www.mantine-react-table.dev/

**Reply by ruslanguns** [reply]
FYI testing on production :-)

**Reply by KevinVandy** [reply]
> FYI testing on production :-)  

Just to be clear though, there's still work to be done on this. It will probably go into beta this weekend though. Docs site should be up tomorrow

**Reply by ruslanguns** [reply]
I will be observing closer the activity in the repository. Thanks

**Comment by hunt3r** [comment]
Hey looks great. I've been using mantine and react-table for about a year. The React-table integration has always been a bit problematic for me. This should be a nice step. 

Is this stable enough to use? Do you foresee any major changes to your interfaces?

**Reply by KevinVandy** [reply]
So the cool thing about this project, is that it was forked from a pretty stable project first: material-react-table.com

I was originally planning on releasing v1.0 stable next week, as it is starting to get stable now (outside of some pending advanced filtering variant features still coming).

Mantine v6.0 is also coming out in a month or so, and I'm weighing if v1.0 should wait for that or not, or if I'm going to just have to release a v2.0 of this library really quickly after releasing v1. Most of those changes will just be around date inputs though.

**Reply by KevinVandy** [reply]
I should say actually, that there are still some TypeScript weirdness for some of the types of mantine props that are still being figured out. They can all be ignored with `as any` for now, I guess.

**Reply by hunt3r** [reply]
I haven't looked into mantine 6 yet, hopefully less breaking change than the move from 4 to 5! I actually just moved from 5.2 to 5.10 today and that went smooth. I'll give your library a try and report back my findings sometime in the next week or two.

**Comment by masterbater** [comment]
I was mixing bootstrap and mui to support material-react-table. Decided to replace bootstrap to mantine. Thanks for making this library with features mirrored from material-react-table.  At first I thought this was not made by the author of material-react-table, but geez @KevinVandy its brilliant of you to make a mantine version. Thanks a lot.

Can we use v0.8.1 in production now?

**Reply by KevinVandy** [reply]
Within the next few days, I'll be moving it from the alpha to beta designation. It probably could have been launched into beta now, but I want to do a much more thorough sweep of all the features. It really is a 99% mirror of material-react-table right now. The main things that will be added before 1.0 have to do with advanced filters and built-in date filters.
